File: data/readable_format/cpp/file1426.cpp
#include <iostream>
#include <cctype> // for std::toupper

// Function declaration
bool OLSAME(char CA, char CB);

int main() {
    // Test cases
    struct TestCase {
        char CA;
        char CB;
        bool expected;
    };

    TestCase tests[] = {
        {'A', 'A', true},
        {'a', 'A', true},
        {'A', 'B', false},
        // Add more test cases as needed
    };

    bool allTestsPassed = true;

    for (const auto& test : tests) {
        bool result = OLSAME(test.CA, test.CB);
        if (result != test.expected) {
            std::cerr << "Test failed: OLSAME(" << test.CA << ", " << test.CB
                      << ") = " << std::boolalpha << result
                      << ", expected: " << test.expected << std::endl;
            allTestsPassed = false;
        }
    }

    if (allTestsPassed) {
        std::cout << "All tests passed." << std::endl;
    } else {
        std::cout << "Some tests failed." << std::endl;
    }

    return 0;
}

// Function definition
bool OLSAME(char CA, char CB) {
    if (CA == CB) return true;
    
    int INTA = static_cast<int>(CA);
    int INTB = static_cast<int>(CB);
    int ZCODE = static_cast<int>('Z');
    
    if (ZCODE == 90 || ZCODE == 122) {
        if (INTA >= 97 && INTA <= 122) INTA -= 32;
        if (INTB >= 97 && INTB <= 122) INTB -= 32;
    } else if (ZCODE == 233 || ZCODE == 169) {
        if ((INTA >= 129 && INTA <= 137) || (INTA >= 145 && INTA <= 153) || (INTA >= 162 && INTA <= 169)) INTA += 64;
        if ((INTB >= 129 && INTB <= 137) || (INTB >= 145 && INTB <= 153) || (INTB >= 162 && INTB <= 169)) INTB += 64;
    } else if (ZCODE == 218 || ZCODE == 250) {
        if (INTA >= 225 && INTA <= 250) INTA -= 32;
        if (INTB >= 225 && INTB <= 250) INTB -= 32;
    }
    
    return INTA == INTB;
}

File: data/readable_format/fortran/file1426.f90
MODULE OLSAMEModule
IMPLICIT NONE
CONTAINS
    LOGICAL FUNCTION OLSAME(CA,CB)
        CHARACTER, INTENT(IN) :: CA, CB
        INTEGER :: INTA, INTB, ZCODE
        OLSAME = CA .EQ. CB
        IF (OLSAME) RETURN
        ZCODE = ICHAR('Z')
        INTA = ICHAR(CA)
        INTB = ICHAR(CB)
        IF (ZCODE .EQ. 90 .OR. ZCODE .EQ. 122) THEN
            IF (INTA .GE. 97 .AND. INTA .LE. 122) INTA = INTA - 32
            IF (INTB .GE. 97 .AND. INTB .LE. 122) INTB = INTB - 32
        ELSE IF (ZCODE .EQ. 233 .OR. ZCODE .EQ. 169) THEN
            IF ((INTA .GE. 129 .AND. INTA .LE. 137) .OR. &
                (INTA .GE. 145 .AND. INTA .LE. 153) .OR. &
                (INTA .GE. 162 .AND. INTA .LE. 169)) INTA = INTA + 64
            IF ((INTB .GE. 129 .AND. INTB .LE. 137) .OR. &
                (INTB .GE. 145 .AND. INTB .LE. 153) .OR. &
                (INTB .GE. 162 .AND. INTB .LE. 169)) INTB = INTB + 64
        ELSE IF (ZCODE .EQ. 218 .OR. ZCODE .EQ. 250) THEN
            IF (INTA .GE. 225 .AND. INTA .LE. 250) INTA = INTA - 32
            IF (INTB .GE. 225 .AND. INTB .LE. 250) INTB = INTB - 32
        END IF
        OLSAME = INTA .EQ. INTB
    END FUNCTION OLSAME
END MODULE OLSAMEModule

PROGRAM TestOLSAME
    USE OLSAMEModule
    IMPLICIT NONE
    CHARACTER :: testChar1, testChar2
    LOGICAL :: result

    ! Test case 1: Same uppercase letters
    testChar1 = 'A'
    testChar2 = 'A'
    result = OLSAME(testChar1, testChar2)
    PRINT *, "Test 1 (A, A): ", result

    ! Test case 2: Same letters, different cases
    testChar1 = 'a'
    testChar2 = 'A'
    result = OLSAME(testChar1, testChar2)
    PRINT *, "Test 2 (a, A): ", result

    ! Test case 3: Different letters
    testChar1 = 'A'
    testChar2 = 'B'
    result = OLSAME(testChar1, testChar2)
    PRINT *, "Test 3 (A, B): ", result

    ! Add more tests as needed...
END PROGRAM TestOLSAME